src/Typeahead.purs

```diff
@@ -1008,18 +1014,26 @@ renderToolbarSearchDropdown defaultLabel resetLabel renderItem renderFuzzy st =
 
 replaceSelected
   :: forall action f item m
-  . Eq item
+  . Data.Foldable.Foldable f
+  => Eq item
   => Effect.Aff.Class.MonadAff m
   => f item
   -> CompositeComponentM action f item m Unit
 replaceSelected selected = do
   st <- Halogen.modify _ { selected = selected }
+  Halogen.tell _multiInput unit
+    $ Ocelot.Components.MultiInput.Component.SetItems
+        ( Data.Array.mapMaybe (Data.Array.head <<< Foreign.Object.values <<< st.itemToObject)
+            <<< Data.Array.fromFoldable
+            $ selected
+        )
```

Comment:
If we use `Insertable _`, we don't have to add the `Data.Foldable.Foldable f` constraint, and we don't have to call `Halogen.tell _multiInput` when we don't have a `MultiInput` mounted.
```purs
case st.insertable of
  NotInsertable -> pure unit
  Insertable toString -> do
    Halogen.tell _multiInput
      $ Ocelot.Components.MultiInput.Component.SetItems (map toString selected)
```

Reply:
Hmm `replaceSelected` is part of `embeddedHandleQuery` in which I don't think we can specialize `item` to `String` though.

Reply:
We don't have to specialize `item` to `String` here, as long as we have an `Insertable item -> String`, which is why I suggested specializing to `String` at the config level. We don't actually have a use-case for a polymorphic `item` when using `MultiInput` yet, so I don't think we'd be losing anything specializing it, but if we wanted to keep `item` polymorphic, we could push the config of `item -> String` to the call-site.

Reply:
There's two conversions we do to use `MultiInput` in `Typeahead`
1. items in `MultiInput` are `String` not `item` so we rely on `itemToObject` which can be interpreted as `item -> Maybe String`
2. selections in `MultiInput` are `Array String` not `f item` so I feel the easiest non-breaking change is to add a `Foldable` constraint which both `Maybe` and `Array` satisfy. We could add a function `f item -> Array item` to `Operations` but it's not used by variants other than `MultiInput` and when we build `MultiInput` on `Select` to manage the same `f item` state we don't need this operator so I don't feel like adding that function.

I'm going to merge this in since you two both approved. We can continue this discussion here or on slack.